util/src/main/java/io/kubernetes/client/util/Yaml.java

```diff
@@ -231,7 +204,58 @@ public static Object load(Reader reader) throws IOException {
    * @throws IOException If an error occurs while reading the YAML.
    */
   public static <T> T loadAs(Reader reader, Class<T> clazz) {
-    return yaml.loadAs(reader, clazz);
+    return getSnakeYaml().loadAs(reader, clazz);
+  }
+
+  /**
+   * Load list of instantiated API objects from a YAML string representation. Returns list of
+   * concrete typed objects (e.g. { V1Pod, V1SERVICE })
+   *
+   * <p>Order of API objects in list will be preserved according to order of objects in YAML string.
+   *
+   * @param content The YAML content
+   * @return List of instantiated objects.
+   * @throws IOException If an error occurs while reading the YAML.
+   */
+  public static List<Object> loadAll(String content) throws IOException {
+    return loadAll(new StringReader(content));
+  }
+
+  /**
+   * Load list of instantiated API objects from a YAML file. Returns list of concrete typed objects
+   * (e.g. { V1Pod, V1SERVICE })
+   *
+   * <p>Order of API objects in list will be preserved according to order of objects in YAML file.
+   *
+   * @param f The file to load.
+   * @return List of instantiated of the objects.
+   * @throws IOException If an error occurs while reading the YAML.
+   */
+  public static List<Object> loadAll(File f) throws IOException {
+    return loadAll(new FileReader(f));
+  }
+
+  /**
+   * Load list of instantiated API objects from a stream of data. Returns list of concrete typed
+   * objects (e.g. { V1Pod, V1SERVICE })
+   *
+   * <p>Order of API objects in list will be preserved according to order of objects in stream of
+   * data.
+   *
+   * @param reader The stream to load.
+   * @return List of instantiated of the objects.
+   * @throws IOException If an error occurs while reading the YAML.
+   */
+  public static List<Object> loadAll(Reader reader) throws IOException {
+    Iterable<Object> iterable = getSnakeYaml().loadAll(reader);
+    List<Object> list = new ArrayList<Object>();
+    for (Object object : iterable) {
+      if (object != null) {
+        list.add(ModelMapper((Map<String, Object>) object));
```

Comment:
No, sorry I was unclear.  The `((Map<String, Object) object)` part of this line could throw ClassCastException (if for example the YAML contains a string) I would catch that and throw a more intelligible error, rather than just letting the class cast percolate up to the caller.

We should keep the `IOException` the same.